Project Home

kpthatsme opened the pull request:
## Changes
This PR introduces the concept of a 'Project Home' behind the feature flag `project-home-exp-5`.

If you're in the bucket:
- There will be a home tab
- You will be taken to '/home' instead of '/insights' if you navigate to app.posthog.com 
- Newly signed up users will be dropped into '/home' instead of '/insights'

Goals are to help activation in a few ways:
- Help new users get back to install instructions
- Provide education content
- Describe types of core insights
- Display recent insights to help start a new analysis

Depending on how this experiment goes there's a lot we will want to iterate on to test and making it more useful for activated users over time (e.g. a module for pinned dashboards, etc).

More details in this experiment here https://github.com/PostHog/experiments/issues/5

![screen-capture (4)](https://user-images.githubusercontent.com/5965891/115928323-c4b22600-a43a-11eb-9c79-3409c892864d.gif)


![screen-capture (5)](https://user-images.githubusercontent.com/5965891/115929880-6175c300-a43d-11eb-8458-ba6dc00ba50d.gif)

<img width="1686" alt="Screen Shot 2021-04-23 at 1 16 14 PM" src="https://user-images.githubusercontent.com/5965891/115929959-81a58200-a43d-11eb-9e11-8efa9e7aad79.png">
<img width="1548" alt="Screen Shot 2021-04-23 at 2 05 22 PM" src="https://user-images.githubusercontent.com/5965891/115929969-8407dc00-a43d-11eb-9899-2007c2eecfcc.png">


## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

Commit:
PH home

Commit:
let data load

Commit:
cleanup

Commit:
add a feature flag

Commit:
add feature flag

Commit:
cleanup

Commit:
css cleanup

Commit:
cleanup

Commit:
copy edits

Commit:
update footer

Commit:
copy edits + open community links in new tab

Commit:
add instrumentation

Commit:
default hasData to false

Commit:
edit rocket css

Commit:
copy edits

kpthatsme commented:
@paolodamico @liyiy @EDsCODE this is ready for review but please don't merge as I'm still testing and may make a couple of minor tweaks

paolodamico commented:
Adding my thoughts here for context. Originally I was thinking we needed to think about this in more detail, I was particularly worried about
1. How this would play out with the changes we're doing on the core querying experience (making sure they both complement each other, not just feel like two random pieces put together)
2. The visuals. I think they require a bit of extra work and input from design to feel consistent with the app.
3. Some of the content. I think we need to be more deliberate on what we show to users to be as helpful as possible. Plus, and more tangential to this, we need to add a ton more documentation/articles on how to do different analytics tasks.
4. The other stuff we had in the previous setup page (which essentially this substitutes), that we'd be losing.


However, after some more thought, I think this totally makes sense as an experiment. In general terms, we're not failing fast enough, and not iterating as much as we should yet. So let's ship this ASAP, learn from it and iterate. With this in mind, I'll jump in with a faster review (in the next hour or so) just to make sure a) the behavior is roughly as expected, b) there aren't any code issues that could affect other parts of the product, and c) make sure we're well instrumented to measure success.

The above being said, I do think that even in the case of very positive results, we should definitely polish this before shipping to more users. Happy to help with this. Thoughts @kpthatsme?

paolodamico commented on frontend/src/scenes/sceneLogic.ts:
```diff
@@ -236,20 +241,37 @@ export const sceneLogic = kea<sceneLogicType>({
                 mapping[path] = (params) => actions.loadScene(scene, params)
             }
         }
+        unmountFeatFlagLogic()
```
Don't think we should unmount this logic, it's used across the app.

paolodamico commented on frontend/src/scenes/sceneLogic.ts:
```diff
@@ -236,20 +241,37 @@ export const sceneLogic = kea<sceneLogicType>({
+
         const mapping: Record<string, (params: Params) => any> = {}
 
+        let experimentRedirect: string | undefined = undefined
```
With this new setup, I will still be unable to reach the new home page until I complete the onboarding process (whether or not I ingest events), is this expected?

kpthatsme replied:
let's chat about this during our sync today i have some questions around the current onboarding

paolodamico commented on frontend/src/scenes/sceneLogic.ts:
```diff
@@ -236,20 +241,37 @@ export const sceneLogic = kea<sceneLogicType>({
+
         for (const [paths, redirect] of Object.entries(redirects)) {
             for (const path of paths.split('|')) {
+                if (path === '/' && redirect) {
```
This doesn't need to run for every redirect path, further the logic here gets convoluted. You just need to check if `path === '/'` and redirect to `/home`. No need to alter this redirect logic.

kpthatsme replied:
Not sure exactly what you mean here – this logic executes only if path === / 

We have a redirect right now that sets / --> /insights – I don't want to alter this rule or the default behavior. Since we are running this as an experiment I only want a subset of those redirects to go to home

Do you have an idea for an alternate approach?

paolodamico replied:
Simplifying logic here in updated commit

paolodamico committed:
very minor tweaks

paolodamico reviewed (changes requested):
In general, behavior seems to work mostly as expected! Made some code comments. Let me know if you need help, and tag me for a review once it's ready again 😉

Commit:
update eventUsage sig

kpthatsme commented on frontend/src/scenes/dashboard/DashboardItem.tsx:
```diff
@@ -251,14 +264,18 @@ export function DashboardItem({
-                                    }
-                                }}
+                                onClick={
+                                    !onClick
```
Context here is we take in an onClick param on this component but weren't actually applying it.

Commit:
change order of default onclick

kpthatsme commented on frontend/src/scenes/onboarding/home/home.tsx:
```diff
@@ -0,0 +1,174 @@
+                    {preflight?.email_service_available ? (
+                        <Button
+                            onMouseLeave={() => {
+                                setShowToolTip(false)
```
Yes unfortunately that wasn't the behavior I was experiencing. I believe cause this tooltip is basically wrapping the entire modal, since it's wrapped around the button which contains the modal (just hidden) so when the modal is active it thinks its div is active too, hence the workarounds

paolodamico replied:
K, no point spending too much time on this right now

Commit:
move invite modal inline

Commit:
add utm param to slack

Commit:
move project home seen call to effect

kpthatsme commented on frontend/src/scenes/onboarding/home/home.tsx:
```diff
@@ -0,0 +1,174 @@
+        </React.Fragment>
+    )
+    const teamHasData = user?.team?.ingested_event
+    reportProjectHomeSeen(teamHasData || false)
```
Agree about adding the effect – not sure I agree re: how we track this / this becoming difficult to keep track of. Going to keep it this way until it is clear there's an issue.

paolodamico replied:
Perfect! For the record it's not so much like there's an issue, but we're adding a custom event for something that could be easily tracked with autocapture

kpthatsme commented on frontend/src/scenes/onboarding/home/home.tsx:
```diff
@@ -0,0 +1,174 @@
+            <div style={{ marginBottom: 128 }} className={'home-container'}>
+                <Space direction="vertical">
+                    <PageHeader
+                        title={`${currentTeam?.name ?? ''} Project Home 🚀`}
```
Updated the name + removed the rocket – re: offbrand, right now we don't have much of a brand within our product. There's a huge disconnect from our marketing site --> product and I think we should find little opportunities to bring personality into the product where there are opportunities to do so.

kpthatsme commented on frontend/src/scenes/onboarding/home/home.tsx:
```diff
@@ -0,0 +1,174 @@
+            <Footer>
+                <div>
+                    <h5>
+                        {`🦔 ❤ `} {currentTeam?.name ?? 'Your team'}
```
Going to just remove the entire footer – I don't think there's value to just showing their team name. Goal of this was similar to my point above we should try to find small ways to incorporate personality into our product to drive affection, esp in these early flows

Commit:
make map inline

kpthatsme commented on frontend/src/scenes/onboarding/home/modules/discoverInsight.tsx:
```diff
@@ -0,0 +1,278 @@
+                    {data.length <= 0 && (
+                        <Space direction={'vertical'}>
+                            <Paragraph style={{ marginTop: 5 }}>
+                                There are no recent analyses. Time to get to work!
```
Don't think there's a lot of value in adding a generic CTA here when there's the row of all insight types directly above this.

Commit:
load data local to component

Commit:
move lessons map inline

Commit:
add insight type to recent analysis event

Commit:
add lesson images to proj

kpthatsme commented on frontend/src/scenes/onboarding/home/modules/posthogLessons.tsx:
```diff
@@ -0,0 +1,153 @@
+            target="_blank"
+            onClick={() => {
+                reportProjectHomeItemClicked('tutorials', lesson.title)
+                return true
```
If you have an onClick on an A tag, the onClick fires first and works as a validation – return true and you follow the link, return false and you don't follow the link. 

This is just so we explicitly follow the link, even if for some reason there's a bug in our event logging.

Commit:
move types to type file

Commit:
update rel if opening in new tab

kpthatsme commented on frontend/src/scenes/sceneLogic.ts:
```diff
@@ -236,20 +241,37 @@ export const sceneLogic = kea<sceneLogicType>({
         ],
     },
     urlToAction: ({ actions }) => {
+        const unmountFeatFlagLogic = featureFlagLogic.mount()
```
Hmmm so that wasn't my experience – without mounting the logic here I was getting a "Can not find path "lib.logic.featureFlagLogic" in the store." error

Commit:
remove unmount

Commit:
fix file names, rename modules

Commit:
cleanup css

Twixes commented:
I think your Git setup doesn't use proper GitHub credentials @kpthatsme, as the PR history looks weird here:

![history](https://user-images.githubusercontent.com/4550621/116399477-4209dc00-a829-11eb-90b3-df0124058dd7.png)

Commit:
switch to ant breakpoints

Commit:
use proper keys

Commit:
update lessons breakpoints

kpthatsme commented:
> Adding my thoughts here for context. Originally I was thinking we needed to think about this in more detail, I was particularly worried about
> 
> 1. How this would play out with the changes we're doing on the core querying experience (making sure they both complement each other, not just feel like two random pieces put together)
> 2. The visuals. I think they require a bit of extra work and input from design to feel consistent with the app.
> 3. Some of the content. I think we need to be more deliberate on what we show to users to be as helpful as possible. Plus, and more tangential to this, we need to add a ton more documentation/articles on how to do different analytics tasks.
> 4. The other stuff we had in the previous setup page (which essentially this substitutes), that we'd be losing.
> 
> However, after some more thought, I think this totally makes sense as an experiment. In general terms, we're not failing fast enough, and not iterating as much as we should yet. So let's ship this ASAP, learn from it and iterate. With this in mind, I'll jump in with a faster review (in the next hour or so) just to make sure a) the behavior is roughly as expected, b) there aren't any code issues that could affect other parts of the product, and c) make sure we're well instrumented to measure success.
> 
> The above being said, I do think that even in the case of very positive results, we should definitely polish this before shipping to more users. Happy to help with this. Thoughts @kpthatsme?


@paolodamico thanks for the review! I've addressed your feedback and it's ready for another pass. 

re: Fulling rolling this out – yes, agreed. Even if this is successful at activating users we should look at this like a prototype to inform future work.

paolodamico commented on frontend/src/scenes/onboarding/home/home.tsx:
```diff
@@ -0,0 +1,174 @@
+                        caption={
+                            !teamHasData ? `Welcome to PostHog! Install one of our libraries to get started.` : ` `
+                        }
+                        buttons={HeaderCTAs()}
```
Keeping this open for future improvements

paolodamico committed:
final tweaks

paolodamico committed:
remove unnecessary extra loads

paolodamico reviewed (approved):
Let's go 🚀🚀 🚀  ! Not merging because you wanted to do it.

A few final notes to keep in mind,
- See my last commit (3984c72f9569f173ba8f16cb1a8123ff2f1e0f16), I removed the extra loads to team and saved insights because we were no longer used them. If this is not intended behavior, please adjust.
- There's some ugly JS errors due to some Ant component in the project home. Not a big deal now but we need to clean it up before expanding the release.

<img width="837" alt="" src="https://user-images.githubusercontent.com/5864173/116466817-d113f580-a823-11eb-97fb-e43ba447deb0.png">

Commit:
move lesson thumbs to public

Commit:
test if adding to pub root works

Commit:
fix file name

Commit:
fix for slick-carousel arrow bug throwing errors

Commit:
move images back to scenes subdir

Commit:
use a component instead of a util

kpthatsme commented:
> Let's go 🚀🚀 🚀 ! Not merging because you wanted to do it.
> 
> A few final notes to keep in mind,
> 
> * See my last commit ([3984c72](https://github.com/PostHog/posthog/commit/3984c72f9569f173ba8f16cb1a8123ff2f1e0f16)), I removed the extra loads to team and saved insights because we were no longer used them. If this is not intended behavior, please adjust.
> * There's some ugly JS errors due to some Ant component in the project home. Not a big deal now but we need to clean it up before expanding the release.
> 
> <img alt="" width="837" src="https://user-images.githubusercontent.com/5864173/116466817-d113f580-a823-11eb-97fb-e43ba447deb0.png">

Thanks! JS error fixed (ended up being a bug in [react-slick](https://github.com/akiran/react-slick/issues/1195).

Commit:
update questions

Commit:
update a word
